### PR TITLE
Update to Unity 5.9

### PR DESCRIPTION
**Pull request by timbussmann**
* The approach to testing calls to `Dispose` on the `UnityContainer` used a wrapper implementation of `IUnityContainer`. However, they changed the logic which configures stuff like our property injection logic to cast `IUnityContainer` to `UnityContainer` which breaks the wrapper approach. `UnityContainer.Dispose` disposes extensions so I've added a canary extension to detect calls to `Dispose`.
* I've tested it with 5.9.0 and that worked. However, till 5.9.4 there seem to be a few patches which make sense to include to I've went with 5.9.4 as the minimum version as later patch versions don't fix anything which should impact our users. Tests run against the latest version (5.9.7 at the time).
* Their built in SingletonLifetimeManager still does not behave the way we need it to (at least according to unit tests), so we need to keep our implementation.

### CODE REVIEW

**Comment by bording** [comment]
Give how often Unity breaks things in arbitrary releases, I don't really see value in setting the minimum version lower than what we are testing. I'd even go so far as to remove the direct package references in the test projects, and rely on the transitive reference from the main project.

**Comment by timbussmann** [comment]
> Give how often Unity breaks things in arbitrary releases, I don't really see value in setting the minimum version lower than what we are testing. I'd even go so far as to remove the direct package references in the test projects, and rely on the transitive reference from the main project.

using transitive dependencies would make it harder to test new releases for breaking changes (where we don't want to release a new version of NServiceBus.Unity every time)?

**Comment by bording** [comment]
> using transitive dependencies would make it harder to test new releases for breaking changes (where we don't want to release a new version of NServiceBus.Unity every time)?

How? You just update the single package reference and run the tests. You can even PR it. It's the same process you'd go through anyway, only you're updating a single reference instead of one for each test project.

**Comment by timbussmann** [comment]
> How? You just update the single package reference and run the tests. You can even PR it. 

in case we're happy to bump the minimum required version every time we test a new version that is true.

**Comment by bording** [comment]
> in case we're happy to bump the minimum required version every time we test a new version that is true.

If we're just testing the new version, we don't actually have to merge the PR if we don't want to do that.

**Comment by timbussmann** [comment]
> If we're just testing the new version, we don't actually have to merge the PR if we don't want to do that.

that's true. I was more thinking about always trying to run the tests against the latest versions but I think that's not a common practice we use for external dependencies though.

**Comment by bording** [comment]
Keeping things as-is would be fine, but I was just thinking about how problematic the Unity updates were vs. trying to keep a wider range of versions we support. For example it seems reasonable to me to make this PR 5.9.7 everywhere, and not 5.9.4 for the actual minimum, because that makes an assumption that 5.9.5 - 5.9.6 also didn't have breaking changes!

For example, what if a break happened in 5.9.5, and the was reverted in 5.9.7? Our tests would still pass, but people in trying to use any of the versions in the middle would be broken.

Keeping the minimum version lower than what we're testing only works if you can reasonably rely on SemVer, and in this case we know we can't.

**Comment by timbussmann** [comment]
the TF agreed that it makes sense to target 5.9.7 as the minimum version. Adjusted the package references and rebased the branch.

File: src/NServiceBus.Unity/PropertyInjectionBuilderStrategy.cs [hunk 1]
```diff
@@ -13,19 +13,19 @@ public PropertyInjectionBuilderStrategy(UnityObjectBuilder unityContainer)
             this.unityContainer = unityContainer;
         }
 
-        public override void PreBuildUp(IBuilderContext context)
+        public override void PreBuildUp(ref BuilderContext context)
```
[the remark this answers is not in the document]
Comment (timbussmann) [reply src/NServiceBus.Unity/PropertyInjectionBuilderStrategy.cs]: 🤷‍♂️ that's a signature change on their base class which we are overriding. Nothing we're able to change.

[the remark this answers is not in the document]
Comment (WilliamBZA) [reply src/NServiceBus.Unity/PropertyInjectionBuilderStrategy.cs]: asd

**Pull request merged**